.github/workflows/deploy.yaml

```diff
@@ -2,18 +2,32 @@ name: Deployment
 
 on:
   workflow_dispatch:
+  pull_request:
+    paths:
+      - "lib/**"
+      - "integration_tests/**"
+      - "package.json"
+      - "package-lock.json"
 
 jobs:
   build_package_and_deploy:
     name: Build, package and deploy
     runs-on: ubuntu-latest
     timeout-minutes: 90
+    permissions:
+      id-token: write
+      contents: read
     env:
-      AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION_DEPLOY }}
-      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID_DEPLOY }}
-      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY_DEPLOY }}
+      AWS_ROLE_ARN: ${{ vars.AWS_ROLE_ARN }}
       AWS_DEFAULT_ACCOUNT: ${{ secrets.AWS_ACCOUNT_ID }}
+      AWS_DEFAULT_REGION: 'us-west-2'
```

Comment:
We might keep this in `us-east-1` or any region but `us-west-2` to avoid bumping into resource quotas.